core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -407,6 +407,24 @@ public ServerBuilder service(PathMapping pathMapping, Service<HttpRequest, HttpR
     }
 
     /**
+     * Decorates and binds the specified {@link ServiceWithPathMapping} at multiple {@link PathMapping}
+     * of the default {@link VirtualHost}.
+     *
+     * @throws IllegalStateException if the default {@link VirtualHost} has been set via
+     *                               {@link #defaultVirtualHost(VirtualHost)} already
+     */
+    public <T extends ServiceWithPathMapping<HttpRequest, HttpResponse>,
+            R extends Service<HttpRequest, HttpResponse>>
+            ServerBuilder service(T serviceWithPathMapping, Function<T, R> decorator) {
+        requireNonNull(serviceWithPathMapping, "serviceWithPathMapping");
+        requireNonNull(serviceWithPathMapping.pathMappings(), "serviceWithPathMapping.pathMappings()");
```

Comment:
Maybe it's better to check it here, because there is a chance when future implementations miss to return a none null List<ParhMapping>. It's almost the same idea with prevent calling this method with null parameters 😉